### PR TITLE
chore(button): deprecate link api features

### PR DESCRIPTION
**Pull request by cdransf**
## Description

Deprecate the link API (`href`, `target`, `download`, `referrerpolicy`, `rel`) on `<sp-button>` and related button components that extend `ButtonBase`. When `href` is set in dev mode, a deprecation warning directs consumers to use native HTML anchor (`<a>`) elements with the `spectrum-Button` class and `@spectrum-web-components/styles/global-elements.css`, the global styling infrastructure established in #6051.

## Motivation and context

Using `<sp-button href="...">` conflates button and link semantics, creating accessibility issues where screen reader users navigating by form controls will not find link-styled buttons. Deprecating the link API reduces the dual link/button maintenance burden and aligns with the Links as Buttons Refactor RFC. This is the next phase following #6051, which established the global styling infrastructure that provides the migration path.

## Related issue(s)

- SWC-1204

## Author's checklist

- [x] I have read the **[CONTRIBUTING](https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)** and **[PULL_REQUESTS](https://github.com/adobe/spectrum-web-components/blob/main/PULL_REQUESTS.md)** documents.
- [x] I have added automated tests to cover my changes.
- [x] I have included updated documentation if my change required it.

---

## Reviewer's checklist

- [x] Includes a Github Issue with appropriate flag or Jira ticket number without a link
- [ ] Automated tests cover all use cases and follow best practices for writing
- [ ] Validated on all supported browsers

### Manual review test cases

- [x] _Verify deprecation warning appears when `href` is set_
  1. Open a page with `<sp-button href="https://example.com">Link</sp-button>`
  2. Open the browser console
  3. Expect a deprecation warning mentioning `href` and referencing `spectrum-Button` class and `global-elements.css`

- [x] _Verify no warning when `href` is not set_
  1. Open a page with `<sp-button>Button</sp-button>`
  2. Open the browser console
  3. Expect no `href`-related deprecation warning

- [x] _Verify existing link functionality still works_
  1. Open a Button story with `href` (e.g., Button > Accent > Fill > href)
  2. Click the rendered link
  3. Expect navigation to occur as before

- [x] _Review README deprecation documentation_
  1. Visit the Button docs page
  2. Check the "Handling events" and "Accessibility" sections
  3. Expect deprecation notices with guidance pointing to native `<a>` elements and the global elements stylesheet

### CODE REVIEW

File: 1st-gen/packages/button/src/ButtonBase.ts [hunk 1]
```diff
@@ -29,6 +29,17 @@ import buttonStyles from './button-base.css.js';
 /**
  * @slot - text content to be displayed in the Button element
  * @slot icon - icon element(s) to display at the start of the button
+ *
+ * The following link-related properties inherited from `LikeAnchor` are
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/src/ButtonBase.ts]: Would it be appropriate to add the `@deprecated` JSDoc tag here?

File: 1st-gen/packages/button/src/ButtonBase.ts [hunk 2]
```diff
@@ -228,10 +239,24 @@ export class ButtonBase extends ObserveSlotText(LikeAnchor(Focusable), '', [
...
+        { level: 'deprecation' }
+      );
+    }
+  }
+
   protected override updated(changed: PropertyValues): void {
     super.updated(changed);
     if (changed.has('href')) {
       this.manageAnchor();
+      if (this.href && this.href.length > 0) {
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/src/ButtonBase.ts:247]: Do we need to account for the other attributes at all? 

If there's an href, there might be a `target` or `download`, etc. But it would never be the other way around right, where there accidentally was just `rel` but no href? It feels a bit to edge-case-y to me, but I figured I could bring it up just in case.

File: 1st-gen/packages/button/test/button.test.ts [hunk 3]
```diff
@@ -86,6 +86,48 @@ describe('Button', () => {
       });
     });
 
+    it('warns in devMode when href is provided', async () => {
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/test/button.test.ts:89]: ❤️

File: 1st-gen/packages/button/README.md [hunk 4]
```diff
@@ -376,14 +398,16 @@ Do not use custom colors for buttons. The colors of different button variations
 
 #### Use a static button-styled native link if including href
 
+> **Deprecated**: The `href` attribute and other link-related properties (`target`, `download`, `referrerpolicy`, `rel`) on `<sp-button>` are deprecated and will be removed in a future release.
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/README.md:401]: We could use the `deprecationNotice` key in the package.json and get a pretty notice on the docs page (this is what I was playing with for action button): 

```
// package.json

...
 "deprecationNotice": "The link API features in @spectrum-web-components/action-button are deprecated and will be removed in a future release. Use a native link element with global Spectrum link styling instead. See the migration guide: https://opensource.adobe.com/spectrum-web-components/migrations/2026-3-16-action-button-link-api-deprecation/"
```

Image

File: 1st-gen/packages/button/stories/index.ts [hunk 5]
```diff
@@ -281,12 +282,22 @@ export const renderMinWidthButton = (props: Properties): TemplateResult => {
 
 const href = 'https://github.com/adobe/spectrum-web-components';
 
+/**
+ * @deprecated The `href` attribute on `<sp-button>` is deprecated.
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/stories/index.ts]: I wonder if we should just remove these stories. If it's demonstrating patterns we no longer really support, is it ok for us to just remove them? 

What do you think?

File: 1st-gen/packages/button/README.md [hunk 6]
```diff
@@ -336,17 +336,39 @@ Events handlers for clicks and other user actions can be registered on a
...
 
-If you intend to create a link with a `href` attribute, we instead offer CSS classes for creating button-styled links. See more information in the [accessibility section](#accessibility).
+##### Link API deprecation
 
-```html demo
+> **Deprecated**: The `href`, `target`, `download`, `referrerpolicy`, and `rel` attributes on `<sp-button>` are deprecated and will be removed in a future release. Use a native HTML anchor (`<a>`) element with the `spectrum-Button` class instead.
+
+Using `<sp-button href="...">` conflates button and link semantics, which creates accessibility issues: screen reader users navigating by form controls will not find link-styled buttons, and vice versa. Native HTML elements provide correct semantics by default.
+
+To migrate, import the global elements stylesheet and apply button classes to native `<a>` elements:
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/README.md]: I think it could still be appropriate to keep some of the old content in this section (it doesn't even have to be exactly right here): 

```suggestion
If you intend to create a link with a `href` attribute, we instead offer CSS classes for creating button-styled links. To migrate, import the global elements stylesheet and apply button classes to native `<a>` elements:f
```

Something about the "If you intend..." part of that documentation is nice, and I like it. To me, it's sort of reminding consumers, "hey if you're planning to do this, pause for a sec a read why you shouldn't." 😆

File: 1st-gen/packages/button/src/ButtonBase.ts (hunk 2 above)
Comment (5t3ph) [reply 1st-gen/packages/button/src/ButtonBase.ts:247]: my two cents is just to worry about `href` as those other properties in and of themselves will not do anything or mean anything semantically on non-links.

File: 1st-gen/packages/button/src/ButtonBase.ts [hunk 7]
```diff
@@ -228,10 +239,24 @@ export class ButtonBase extends ObserveSlotText(LikeAnchor(Focusable), '', [
     this.addEventListener('keypress', this.handleKeypress);
   }
 
+  private warnLinkAPIDeprecation(): void {
+    if (window.__swc?.DEBUG) {
+      window.__swc.warn(
+        this,
+        `The "href" attribute on <${this.localName}> is deprecated and will be removed in a future release. Use a native HTML anchor (<a>) element with the "spectrum-Button" class instead. Import "@spectrum-web-components/styles/global-elements.css" to enable button styling on native elements.`,
```
Comment (5t3ph) [inline 1st-gen/packages/button/src/ButtonBase.ts]: A note that classes for Action Button will also be available (my current ticket).

File: 1st-gen/packages/button/stories/index.ts (hunk 5 above)
Comment (5t3ph) [reply 1st-gen/packages/button/stories/index.ts]: I suggest keeping them - it exposes the classes as a reference and re-enforces the correct pattern post-deprecation.

File: 1st-gen/packages/button/README.md [hunk 8]
```diff
@@ -336,17 +336,39 @@ Events handlers for clicks and other user actions can be registered on a
...
   class="spectrum-Button spectrum-Button--accent"
-  href="https://github.com/adobe/spectrum-web-components"
+  href="https://opensource.adobe.com/spectrum-web-components"
 >
-  Accent link button
+  Visit docs
 </a>
 ```
 
+See the [accessibility section](#use-a-static-button-styled-native-link-if-including-href) for the full list of supported styling classes and placement requirements.
```
Comment (5t3ph) [inline 1st-gen/packages/button/README.md]: "...for more details" is more accurate since it doesn't list the classes.

File: 1st-gen/packages/button/README.md [hunk 9]
```diff
@@ -336,17 +336,39 @@ Events handlers for clicks and other user actions can be registered on a
...
 
-If you intend to create a link with a `href` attribute, we instead offer CSS classes for creating button-styled links. See more information in the [accessibility section](#accessibility).
+##### Link API deprecation
 
-```html demo
+> **Deprecated**: The `href`, `target`, `download`, `referrerpolicy`, and `rel` attributes on `<sp-button>` are deprecated and will be removed in a future release. Use a native HTML anchor (`<a>`) element with the `spectrum-Button` class instead.
+
+Using `<sp-button href="...">` conflates button and link semantics, which creates accessibility issues: screen reader users navigating by form controls will not find link-styled buttons, and vice versa. Native HTML elements provide correct semantics by default.
+
+If you intend to create a link with a `href` attribute, we instead offer CSS classes for creating button-styled links. To migrate, import the global elements stylesheet and apply button classes to native `<a>` elements:f
```
Comment (5t3ph) [inline 1st-gen/packages/button/README.md]: Typo!
```suggestion
If you intend to create a link with a `href` attribute, we instead offer CSS classes for creating button-styled links. To migrate, import the global elements stylesheet and apply button classes to native `<a>` elements:
```

File: 1st-gen/packages/button/stories/index.ts (hunk 5 above)
Comment (cdransf) [reply 1st-gen/packages/button/stories/index.ts]: ~~That makes sense — if the pattern is deprecated we don't need the stories. ✨~~

File: 1st-gen/packages/button/src/ButtonBase.ts (hunk 1 above)
Comment (cdransf) [reply 1st-gen/packages/button/src/ButtonBase.ts]: I _believe_ that would read like the whole class is deprecated, no? ✨

File: 1st-gen/packages/button/src/ButtonBase.ts (hunk 7 above)
Comment (cdransf) [reply 1st-gen/packages/button/src/ButtonBase.ts]: Got it! I've added a `TODO` for that. ✨

File: 1st-gen/packages/button/README.md [hunk 10]
```diff
@@ -336,17 +336,39 @@ Events handlers for clicks and other user actions can be registered on a
 <sp-button onclick="spAlert(this, '<sp-button> clicked!')">Click me</sp-button>
 ```
 
-If you intend to create a link with a `href` attribute, we instead offer CSS classes for creating button-styled links. See more information in the [accessibility section](#accessibility).
+##### Link API deprecation
```
Comment (marissahuysentruyt) [inline 1st-gen/packages/button/README.md]: What would you think about just keeping this as an h4? Does that make sense still from a hierarchy standpoint? 

That text is just so tiny! 😆 🤏

**Comment by cdransf** [comment]
> I mainly had questions! A more general question I had was if we need to do anything deprecation-wise to `Button.ts`. I had started some of the next ticket related to action button, which also extends ButtonBase, but was rummaging around in ActionButton.ts, adding the deprecation warning there.

There shouldn't be! `Button` will inherit the deprecation warning from `ButtonBase` unless it's overridden (so I think we wouldn't need it in `ActionButton` unless we want a distinct message?) ✨ 

> Also, do we need to create a more "formal" migration guide? I see we have a "[Migrations](https://github.com/adobe/spectrum-web-components/tree/main/1st-gen/projects/documentation/content/migrations)" directory, but it doesn't look like we've added anything there since 1.0.

I'd be inclined to do a migration guide when the properties are removed but, since we're issuing a warning, I don't think we need one for now. ✨

File: 1st-gen/packages/button/README.md (hunk 10 above)
Comment (cdransf) [reply 1st-gen/packages/button/README.md]: Sounds good to me ✨

File: 1st-gen/packages/button/src/ButtonBase.ts (hunk 1 above)
Comment (marissahuysentruyt) [reply 1st-gen/packages/button/src/ButtonBase.ts]: I'm not sure. If you have doubts about it though, I'll defer to you!

**Review by marissahuysentruyt — APPROVED** [review]
~~Do we need to add a changeset for this? We didn't really change the shipped code, but would it be helpful for consumers to see that the deprecations are coming in the changelog?~~

Looking good! I have some notes for the action button (like those TODOs), but I think this is looking great. 👍
